File: 2239 - Find Closest Number to Zero/closestToZero.py
class Solution:
    def findClosestNumber(self, nums):
        min_distance = 10 ** 5 + 1

        for i in range(len(nums)):
            if (abs(nums[i]) < abs(min_distance)):
                min_distance = nums[i]
        
        if (min_distance < 0 and (abs(min_distance) in nums)):
            return abs(min_distance)
        
        return min_distance


if __name__ == '__main__':
    solution = Solution()
    nums1 = [-4,-2,1,4,8]
    nums2 = [2,-1,1]
    print(solution.findClosestNumber(nums1)) # Output: 1
    print(solution.findClosestNumber(nums2)) # Output: 1

File: 2239 - Find Closest Number to Zero/closestToZero.java
class Solution{
    public static int findClosestNumber(int[] nums){
        int minDistance = Integer.MAX_VALUE;

        for(int i = 0; i < nums.length; ++i){
            if(Math.abs(nums[i]) < Math.abs(minDistance)){
                minDistance = nums[i];
            }  
        }
        if(minDistance < 0 && presence(Math.abs(minDistance), nums)){
            return Math.abs(minDistance);
        }
        return minDistance;
    }

    public static boolean presence(int value, int[] nums){
        for(int num : nums){
            if (num == value){
                return true;
            }
        }
        return false;
    }
    
    public static void main(String args[]){
        int[] nums1 = {-4, -2, 1, 4, 8};
        int[] nums2 = {2, -1, 1};
        
        System.out.println(findClosestNumber(nums1));
        System.out.println(findClosestNumber(nums2));
    }
}